mlflow/models/model.py

```diff
@@ -29,55 +30,140 @@
 _MLFLOW_VERSION_KEY = "mlflow_version"
 
 
-class ModelInfo(NamedTuple):
+class ModelInfo:
     """
     The metadata of a logged MLflow Model.
     """
 
-    #: Run relative path identifying the logged model.
-    artifact_path: str
+    def __init__(
+        self,
+        artifact_path: str = None,
+        flavors: Dict[str, Any] = None,
+        model_uri: str = None,
+        model_uuid: str = None,
+        run_id: str = None,
+        saved_input_example_info: Optional[Dict[str, Any]] = None,
+        signature_dict: Optional[Dict[str, Any]] = None,
+        signature=None,  # Optional[ModelSignature]
+        utc_time_created: str = None,
+        mlflow_version: str = None,
```

Comment:
```suggestion
        artifact_path: str,
        flavors: Dict[str, Any],
        model_uri: str,
        model_uuid: str,
        run_id: str = None,
        saved_input_example_info: Optional[Dict[str, Any]],
        signature_dict: Optional[Dict[str, Any]],
        signature,  # Optional[ModelSignature]
        utc_time_created: str,
        mlflow_version: str,
```

Do we need the default values?

Reply:
In general I guess not, but for signature_dict as it will be deprecated I suggest we keep default value None for it :)